src/libasr/codegen/wasm_to_x86.cpp

```diff
@@ -129,23 +130,59 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
 
     void visit_EmtpyBlockType() {}
 
+    void visit_Br(uint32_t label_index) {
+        // Branch is used to jump to the `loop.head` or `loop.end`.
+        if (if_unique_id.size() - loop_unique_id.size() == label_index - 1) {
+            // cycle/continue or loop.end
+            m_a.asm_jmp_label(".loop.head_" + loop_unique_id.back());
+        } else {
+            // exit/break
+            m_a.asm_jmp_label(".loop.end_" + loop_unique_id.back());
+        }
```

Comment:
This `if-else` condition inside the `visit_Br()` is not clear to me. @Thirumalai-Shaktivel, please, could you possibly share how this `if-else` works and chooses whether to iterate the loop or end it?

Reply:
Consider this example:
```py
while i < 10:
    j = 0
    while j < 10:
        k += 1
        if i == 0:
            if j == 3:
                continue
            else:
                j += 1
        j += 1
    i += 1
assert k == 95
```
Output printed using:
```diff
diff --git a/src/libasr/codegen/wasm_to_x86.cpp b/src/libasr/codegen/wasm_to_x86.cpp
index 9db9d2b4b..48f947076 100644
--- a/src/libasr/codegen/wasm_to_x86.cpp
+++ b/src/libasr/codegen/wasm_to_x86.cpp
@@ -131,6 +131,9 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
     void visit_EmtpyBlockType() {}
 
     void visit_Br(uint32_t label_index) {
+std::cout << "label_index: " << label_index << "\n";
+std::cout << "Loop_size: " << loop_unique_id.size() << "\n";
+std::cout << "If_size: " << if_unique_id.size() << "\n";
         // Branch is used to jump to the `loop.head` or `loop.end`.
```
```py
$ lpython examples/expr2.py --backend wasm_x86 && ./expr2.out
label_index: 3
Loop_size: 2
If_size: 4
label_index: 1
Loop_size: 2
If_size: 2
label_index: 1
Loop_size: 1
If_size: 1
```
We have 2 loop's and 2 if's. But in the backend, we get 2 loop's and 4 if's, right?
The first br comes from the continue, if we substract 4 if's with 2 loop's: we get the actual 2 if's. so the br label is 2+1. and So on...
Any doubts, please ask

Reply:
> how this if-else works and chooses whether to iterate the loop or end it?

I didn't notice, how the loop is being exited. I will look into it soon.

Reply:
I see, the Loop is exited through the else block:
```asm
.loop.head_297:
    mov eax, [ebp-4]
    push eax
    push 0x0000000a
    pop ebx
    pop eax
    cmp eax, ebx
    jl .compare_1303
    push 0x00
    jmp .compare.end_303
.compare_1303:
    push 0x01
.compare.end_303:
    pop eax
    cmp eax, 0x01
    je .then_304
    jmp .else_304
.then_304:
    mov eax, [ebp-4]
    push eax
    push 0x00000001
    pop ebx
    pop eax
    add eax, ebx
    push eax
    pop eax
    mov [ebp-4], eax
    jmp .loop.head_297
    jmp .endif_304
.else_304:
.endif_304:
.loop.end_297:
    mov esp, ebp
    pop ebp
    ret
```

Reply:
I have a doubt, here the jump and label `.endif_304` is useless? is it ok? @Shaikh-Ubaid

Reply:
@Shaikh-Ubaid, where you able to understand or get the point?

Reply:
>I have a doubt, here the jump and label .endif_304 is useless? is it ok? @Shaikh-Ubaid

I think for the moment that should be fine.

>@Shaikh-Ubaid, were you able to understand or get the point?

Yup, thank you so much for the explanation, Thirumalai.

Reply:
> It seems this condition (if_unique_id.size() - loop_unique_id.size() == label_index - 1) might not be perfect to determine to continue or end the loop. 

+1, I agree with this.
Also, I didn't test loop within if statement.
There you go, you found the actual bug in the loop statement.
Thank you for testing this. I will create this a new issue. Let's work on this and fix it.